r/inst/build_arrow_static.sh

```diff
@@ -55,6 +55,13 @@ else
   ARROW_DEFAULT_PARAM="OFF"
 fi
 
+# Disable mimalloc on IntelLLVM
```

Comment:
Could you add a comment why we need to disable mimalloc? (Or could you open a new issue for it and refer it in this comment?)

Reply:
done